### DISCUSSION TITLE
"Holding Button1 for 40 seconds to factory reset" does not work on Sonoff 4CHPRO R3?

### DISCUSSION THREAD

**Discussion by whc2001** [arendst/Tasmota#11925] [Q&A] [answered]
I need to factory reset a Sonoff 4CHPRO R3 for networking change. I tried to hold button 1, but relay 1 immediately kicks in right after I press the button down. About 2 seconds later the red status LED flashes once, and 2 more seconds later flashes another twice. I keep holding it down after those flashes, but it does not do anything further. If I release the button relay 1 shuts off. Device is not reset nor rebooted nonetheless. I wonder if this functions was removed or disabled on Sonoff 4CHPRO R3?

**Comment by sfromis** [comment, marked as answer]
Sounds like you may be using [SetOption13](https://tasmota.github.io/docs/Commands/#setoption13) to enforce immediate response, I'd sort of expect that to disable the 40-second hold recovery feature.

**Reply by whc2001** [reply]
I don't think I ever set that option. Do I need to set it to 1 for the long press reset to work?

**Reply by sfromis** [reply]
You have done something to make the relay kick immediately after pressing the button down. That is not default. Do check if you have set the option to ON/1, which is what I think could disable the recovery interval. The point is that 40 seconds is a "very long press", and immediate action disables [long press](https://tasmota.github.io/docs/Buttons-and-Switches/#long-press).

**Reply by whc2001** [reply]
I checked the Wiki and listed those options that might influence the long press:
```
10:57:50.822 CMD: ButtonTopic
10:57:50.827 RSL: RESULT = {"ButtonTopic":"0"}
10:57:56.107 CMD: SetOption1
10:57:56.113 RSL: RESULT = {"SetOption1":"OFF"}
10:58:00.373 CMD: SetOption11
10:58:00.378 RSL: RESULT = {"SetOption11":"OFF"}
10:58:03.314 CMD: SetOption13
10:58:03.318 RSL: RESULT = {"SetOption13":"OFF"}
10:58:09.031 CMD: SetOption32
10:58:09.035 RSL: RESULT = {"SetOption32":40}
```
Seems like everything is default. I wonder if there is something else that also preventing long press event?

**Reply by sfromis** [reply]
There seems to be special handling around Sonoff 4Ch Pro... From docs about [modules](https://tasmota.github.io/docs/Modules/):
"Button handling disregarding SetOption13 only allowing single press to enable RF learning while holding the button".

**Reply by whc2001** [reply]
Oh so I can only use the multiple power cycle method to reset it? Actually I want to use that as a last resort as the wire also has other devices on it. Speaking of RF learning, any information on that? I do need to pair the RM433 remote to it, but I can't find any instruction on the Wiki.

**Reply by sfromis** [reply]
Tasmota is just not involved in the RF part. The information should be in the docs from iTead.

**Reply by whc2001** [reply]
Okay so I reset the device using multiple power cycle method finally. And you are right I found the instruction on the manual and paired the remote successfully. Thanks very much.

**Reply by barbudor** [reply]
1 to 4 button press are already assigned to relay toggle
6 button press starts wifi manager
So it should be possible to create a Rule to capture 5 buttons press to trigger a `reset 1` command
